cluster_management/src/main/java/com/pinterest/rocksplicator/LeaderFollowerStateModelFactory.java

```diff
@@ -313,7 +313,7 @@ public void onBecomeLeaderFromFollower(Message message, NotificationContext cont
         LOG.error("[" + dbName + "] Getting external view");
         view = Utils.getHelixExternalViewWithFixedRetry(admin, cluster, resourceName);
         LOG.error("[" + dbName + "] Got external view");
-        stateMap = view.getStateMap(partitionName);
+        stateMap = Utils.getStateMap(view, partitionName);
```

Comment:
behavior change:

this is also part of follower->leader transition, where the leader tries to inform all non-leaders about it's leadership, so the followers can set upstream correctly. It find out the non-leaders by contacting the stateMap.

Currently the behavior is a NPE and cause the leader transition to fail (ERROR shard). 

The new behavior would be to let the leader transition go through. I think this is preferred, we need the leader for the shard to up, and informing other non-leaders should be a best-effort, especially in this case there is none available. (we also have logic to reset upstream in runtime even if a wrong upstream is set)